sec1_intro_and_setup/test_app/bin/webpack

```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env ruby
```

Comment:
Whats this file for?

Reply:
Rails use `webpack` to bundle Javascripts. This file requires necessary setup to run `webpack`.

Reply:
1. What is webpack?
2. How does it work?

Reply:
1. `Webpack` is a module bundler that bundles Javascripts in Rails.
2.  `Webpack` packs your code in a convenient way for browsers by internally building a dependency graph depends on how your code is used in your apps so that one can easily manage, debug, verify, and test their code.

Reply:
1. What is a `module bundler` ?
2. Is that all webpack does? if it is, why did rails move to webpack?
3. What was rails using before webpack?

Reply:
1.  A `module bundler` is a tool that takes pieces of JavaScript and their dependencies and bundles them into a single file.
2. I mention some reasons that seem to have a bigger impact on switching to `webpack`:
    - `webpack` has some advanced features not available in Sprockets such as modularity.
    - To adopt ES6 syntax
3.  `Sprockets`

Reply:
So, why webpack and not others?

Reply:
Webpack supports modular development and component-based development which Grunt, Gulp doesn't have. Although Parcel is a potential packaging tool to replace Webpack, it does not have a large community as Webpack which makes solution searching difficult when you encounter a problem.